### PR TITLE
Ruff 0.5

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Feature branch for Ruff 0.5. 

### Changes

* https://github.com/astral-sh/ruff/pull/9814
* https://github.com/astral-sh/ruff/pull/11697
* https://github.com/astral-sh/ruff/pull/11194
* https://github.com/astral-sh/ruff/pull/11115
* https://github.com/astral-sh/ruff/pull/10172
* https://github.com/astral-sh/ruff/pull/12011
* https://github.com/astral-sh/ruff/pull/12006
* https://github.com/astral-sh/ruff/pull/10497
* https://github.com/astral-sh/ruff/pull/10667
* https://github.com/astral-sh/ruff/pull/12027
* https://github.com/astral-sh/ruff/pull/12021
* https://github.com/astral-sh/ruff/pull/9559
* https://github.com/astral-sh/ruff/pull/12010
* https://github.com/astral-sh/ruff/pull/10416
* https://github.com/astral-sh/ruff/pull/11901
* https://github.com/astral-sh/ruff/pull/11902
* https://github.com/astral-sh/ruff/pull/12059

### Stabilization

* https://github.com/astral-sh/ruff/pull/12026
* https://github.com/astral-sh/ruff/pull/12029
* https://github.com/astral-sh/ruff/pull/12045
* https://github.com/astral-sh/ruff/pull/12043
* https://github.com/astral-sh/ruff/pull/12049
* https://github.com/astral-sh/ruff/pull/12048
* https://github.com/astral-sh/ruff/pull/12047
* https://github.com/astral-sh/ruff/pull/11220
* https://github.com/astral-sh/ruff/pull/12051
* https://github.com/astral-sh/ruff/pull/12050

### Issues

* https://github.com/astral-sh/ruff/issues/7349
* https://github.com/astral-sh/ruff/issues/11097
* https://github.com/astral-sh/ruff/pull/11115
* https://github.com/astral-sh/ruff/issues/7992
* https://github.com/astral-sh/ruff/issues/10342
* https://github.com/astral-sh/ruff/issues/9994
* https://github.com/astral-sh/ruff/issues/8122
* https://github.com/astral-sh/ruff/issues/6465
* https://github.com/astral-sh/ruff/issues/7349
* https://github.com/astral-sh/ruff/issues/10303
* https://github.com/astral-sh/ruff/issues/8447

> **WARN:** Don't squash merge this PR. 

## TODO

* ~~Drop the empty *Start Ruff 0.5* commit~~

### CODE REVIEW

**Comment by Skylion007** [comment]
Any rules are planning upgrading to stable? Like some of the refurb rules?

**Comment by charliermarsh** [comment]
Yeah we do plan to stabilize some rules in this release.

File: Cargo.toml [hunk 1]
```diff
@@ -219,3 +219,60 @@ opt-level = 1
...
+  "i686-unknown-linux-gnu",
+  "i686-unknown-linux-musl",
+  "powerpc64-unknown-linux-gnu",
+  "powerpc64le-unknown-linux-gnu",
+  "s390x-unknown-linux-gnu",
+  "x86_64-apple-darwin",
+  "x86_64-pc-windows-msvc",
+  "x86_64-unknown-linux-gnu",
+  "x86_64-unknown-linux-musl",
+]
```
Comment (MichaReiser) [inline Cargo.toml:258]: @charliermarsh does ruff now support more targets than before? ;)

**Review by dhruvmanila — APPROVED** [review]
🚀

**Pull request merged**